spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -2428,21 +2429,16 @@ public void doSend(Channel channel, String exchangeArg, String routingKeyArg, Me
 
 	protected void observeTheSend(Channel channel, Message message, boolean mandatory, String exch, String rKey) {
 
-		if (!this.observationRegistryObtained) {
+		if (!this.observationRegistryObtained && this.observationEnabled) {
 			obtainObservationRegistry(this.applicationContext);
 			this.observationRegistryObtained = true;
 		}
-		Observation observation;
 		ObservationRegistry registry = getObservationRegistry();
-		if (!this.observationEnabled || registry == null) {
-			observation = Observation.NOOP;
-		}
-		else {
-			observation = RabbitTemplateObservation.TEMPLATE_OBSERVATION.observation(this.observationConvention,
-					DefaultRabbitTemplateObservationConvention.INSTANCE,
+		Observation observation = RabbitTemplateObservation.TEMPLATE_OBSERVATION.observation(this.observationConvention,
+				DefaultRabbitTemplateObservationConvention.INSTANCE,
+					(Supplier<RabbitMessageSenderContext>) () ->
```

Comment:
With the latest Observation SNAPSHOT we don't need an explicit cast - they just remove the method with `Context`. Just `Supplier` is possible from now on.